packages/@angular/cli/commands/add.ts

```diff
@@ -0,0 +1,102 @@
+import chalk from 'chalk';
+import { Command, CommandScope, Option } from '../models/command';
+import { parseOptions } from '../models/command-runner';
+import { CliConfig } from '../models/config';
+import { SchematicAvailableOptions } from '../tasks/schematic-get-options';
+
+const SilentError = require('silent-error');
+
+
+export default class AddCommand extends Command {
+  readonly name = 'add';
+  readonly description = 'Add support for a library to your project.';
+  scope = CommandScope.inProject;
+  arguments = ['collection'];
+  options: Option[] = [];
+
+  private _rawArgs: any;
+
+  async initializeRaw(args: any): Promise<any> {
+    this._rawArgs = args.slice();
+    return args;
+  }
+
+  public async parseOptions(collectionName: string): Promise<any> {
+    const SchematicGetOptionsTask = require('../tasks/schematic-get-options').default;
+
+    const getOptionsTask = new SchematicGetOptionsTask({
+      ui: this.ui,
+      project: this.project
+    });
+
+    const availableOptions: SchematicAvailableOptions[] = await getOptionsTask.run({
+      schematicName: 'ng-add',
+      collectionName,
+    });
+    let anonymousOptions: string[] = [];
+
+    if (availableOptions) {
+      const nameOption = availableOptions.filter(opt => opt.name === 'name')[0];
+      if (nameOption) {
+        anonymousOptions = [...anonymousOptions, 'name'];
+      }
+    } else {
+      anonymousOptions = [...anonymousOptions, 'name'];
+    }
+
+    this.arguments = this.arguments.concat(anonymousOptions);
+    this.options = this.options.concat(availableOptions || []);
+
+    return parseOptions(this._rawArgs, this.options, this.arguments);
+  }
+
+  async run(commandOptions: any) {
+    const collectionName = commandOptions.collection;
+
+    if (!collectionName) {
```

Comment:
This logic should live in the `validate` function.

Reply:
Done.